File: Java/Strings/Easy/CheckPalindrome.java
// Check Palindrome
// Problem Statement

// Given a string s, determine if it is a palindrome.
// A palindrome reads the same forward and backward. Ignore case sensitivity.

// Example:
// Input: "Racecar" → Output: true
// Input: "hello" → Output: false
package Java.Strings.Easy;

public class CheckPalindrome {
    public static void main(String[] args) {
        String s = "Racecar";
        System.out.println(s + " is a palindrome. " +  isPalindrome(s));
    }

    public static boolean isPalindrome(String s) {
        s = s.toLowerCase();
        int left = 0, right = s.length() - 1;
        while (left < right) {
            if (s.charAt(left) != s.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: Swift/Strings/CheckPalindrome.swift
func isPalindrome(_ s: String) -> Bool {
    let chars = Array(s.lowercased())
    var left = 0, right = chars.count - 1
    while left < right {
        if chars[left] != chars[right] {
            return false
        }
        left += 1
        right -= 1
    }
    return true
}

// Example
print(isPalindrome("Racecar")) // true
print(isPalindrome("hello"))   // false
